packages/web/components/Dashboard/Seed.tsx

```diff
@@ -1,36 +1,303 @@
+};
+
+type MarketProps = {
+  identifier: string;
+};
+
+export const Seed: FC = () => {
+  const [token, setToken] = useState<TokenDataProps | null>();
+
+  useEffect(() => {
+    if (typeof token !== 'undefined') return;
+    (async () => {
+      try {
+        const tokenResponse = await fetch(
+          `${apiUrl}coins/${tokenId + tokenQuery}`,
+        );
+        const tokenJson = await tokenResponse.json();
+        const chartResponse = await fetch(
+          `${apiUrl}coins/${tokenId + chartQuery}`,
+        );
+        const chartJson = await chartResponse.json();
+
+        const { market_data, tickers } = tokenJson;
+        const { prices, total_volumes } = chartJson;
+
+        const buildTokenData = () => {
+          const tokenData: TokenDataProps = {
+            market: market_data,
+            ticker: ticker(tickers, 'balancer_v1'),
+            priceUp:
+              priceIncreased(market_data.price_change_percentage_24h) > 0,
+            volumeUp: volIncreased(total_volumes, market_data.total_volume) > 0,
+            volumePercent: volumeChange(
+              total_volumes,
+              market_data.total_volume,
+            ).toFixed(2),
+            highLow7d: findHighLowPrice(prices, 7),
+            prices,
+          };
+
+          if (!token) {
+            return tokenData;
+          }
+          return null;
+        };
+        setToken(() => buildTokenData());
+        return () => {};
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.log('getTokenData: ', error);
+        return null;
+      }
+    })();
+  });
+
+  return (
+    <>
+      <Box position="relative" zIndex="20">
+        <StatGroup position="relative" my={5} zIndex={10}>
+          <Stat mb={3}>
+            <StatLabel>Market Price</StatLabel>
+            <StatNumber>${token?.market.current_price.usd}</StatNumber>
+            <StatHelpText>
+              <StatArrow type={token?.priceUp ? 'increase' : 'decrease'} />
+              {`${token?.market.price_change_percentage_24h?.toFixed(2)}%` ||
+                'not enough data'}
+            </StatHelpText>
+          </Stat>
+
+          <Stat mb={3}>
+            <StatLabel>24h Trading Volume</StatLabel>
+            <StatNumber>${token?.market.total_volume.usd}</StatNumber>
+            <StatHelpText>
+              <StatArrow type={token?.volumeUp ? 'increase' : 'decrease'} />
+              {token?.volumePercent}%
+            </StatHelpText>
+          </Stat>
+
+          <Stat alignSelf="flex-start" flex="0 0 100%">
+            <StatLabel>7d Low / High</StatLabel>
+            <StatNumber>
+              ${token?.highLow7d.low} / ${token?.highLow7d.high}
+            </StatNumber>
+          </Stat>
+        </StatGroup>
+      </Box>
+      <Box
+        className="chartWrapper"
+        position="absolute"
+        width="100%"
+        height="100%"
+        bottom={0}
+        left={0}
+        zIndex={0}
+        sx={{
```

Comment:
I guess it would clean things up a bit more if you'ld put these styles in a const somewhere and just put the const here